sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb

```diff
@@ -141,7 +141,7 @@ def internal_start_span(name, kind, attributes, links, start_timestamp, parent_c
           if result.recording? && !@stopped
             trace_flags = result.sampled? ? OpenTelemetry::Trace::TraceFlags::SAMPLED : OpenTelemetry::Trace::TraceFlags::DEFAULT
             context = OpenTelemetry::Trace::SpanContext.new(trace_id: trace_id, span_id: span_id, trace_flags: trace_flags, tracestate: result.tracestate)
-            attributes = attributes&.merge(result.attributes) || result.attributes
+            attributes = attributes&.merge(result.attributes) || result.attributes.dup
```

Comment:
Previously, this hash was duped inside `Span.new` (described above), but it isn't anymore, so it's dup'ed here instead.